src/v/cluster/data_migration_backend.h

```diff
@@ -104,6 +104,7 @@ class backend {
     on_partition_work_completed(model::ntp&& ntp, id migration, state state);
 
     /* deferred event handlers */
+    // call only with _mutex lock grabbed
```

Comment:
@mmaslankaprv I decided against passing mutex units reference all the way through. Compiler wouldn't optimize it away, so my mind would remain laden with the sin till the rest of of my days.